l2l/optimizers/kalmanfilter/enkf.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+from l2l.optimizers.kalmanfilter.kalman_utils import _get_shapes, \
+    _encode_targets, _get_batches
+
+try:
+    from numba import jit
+except ModuleNotFoundError:
+    # create a mock decorator if `numba` is not found
+    def jit(**kwargs):
+        def decorator(f):
+            def wrapper(*args, **kwargs):
+                return f(*args, **kwargs)
+            return wrapper
+        return decorator
+
+
+class EnsembleKalmanFilter:
+    def __init__(self, maxit=1, n_batches=1, online=False):
+        """
+        Ensemble Kalman Filter (EnKF)
+
+        EnKF following the formulation found in Iglesias et al. (2013),
+        The Ensemble Kalman Filter for Inverse Problems.
+        doi:10.1088/0266-5611/29/4/045001
+
+        :param maxit: int, maximum number of iterations
+        :param n_batches, int,  number of batches to used in mini-batch. If set
+            to `1` uses the whole given dataset. Default is `1`.
+        :param online, bool, True if one random data point is requested,
+            between [0, dims], otherwise do mini-batch. `dims` is the number of
+            observations. Default is False
+        """
+        self.ensemble = None
+        self.observations = None
+
+        self.maxit = maxit
```

Comment:
I think maxit should be changed to the full name max_iterations. Do we need to document the difference between these iterations and the iterations from the optimizer itself? I think the distinction is not very clear... maybe there is no distinction at all :)